Analytics-CSharp/Segment/Analytics/Configuration.cs

```diff
@@ -49,12 +52,13 @@ public class Configuration
         /// <param name="exceptionHandler">set an exception handler to handle errors happened in async methods within the analytics scope</param>
         /// <param name="storageProvider">set a storage provide to tell the analytics where to store your data:
         ///     <list type="bullet">
-        ///         <item><description><see cref="InMemoryStorageProvider"/> stores data only in memory</description></item>
+        ///         <item><description><see cref="InMemoryStorageProvider"/> stores data only in memory and ignores the persistentDataPath</description></item>
         ///         <item><description><see cref="DefaultStorageProvider"/> persists data in local disk. This is used by default</description></item>
         ///     </list>
+        ///     defaults to DefaultStorageProvider on Unity (Mono) and Xamarin, or to InMemoryStorageProvider on .Net Core
         /// </param>
         public Configuration(string writeKey,
-            string persistentDataPath,
+            string persistentDataPath = null,
```

Comment:
now that we have a way to provide `StorageProvider`, I think we can move this parameter to where we create a storage provider. since these two things are related, it makes sense to put them at one place.